src/methods/wallet-get-calls-status.ts

```diff
@@ -0,0 +1,85 @@
+import { rpcErrors } from '@metamask/rpc-errors';
+import type { Infer } from '@metamask/superstruct';
+import {
+  nonempty,
+  optional,
+  mask,
+  string,
+  array,
+  object,
+  tuple,
+} from '@metamask/superstruct';
+import type {
+  Json,
+  JsonRpcRequest,
+  PendingJsonRpcResponse,
+} from '@metamask/utils';
+import { HexChecksumAddressStruct, StrictHexStruct } from '@metamask/utils';
+
+import { validateParams } from '../utils/validation';
+
+const GetCallsStatusStruct = tuple([nonempty(string())]);
+
+const GetCallsStatusReceiptStruct = object({
+  logs: array(
+    object({
+      address: HexChecksumAddressStruct,
+      data: StrictHexStruct,
+      topics: array(StrictHexStruct),
+    }),
+  ),
+  status: StrictHexStruct,
+  chainId: optional(StrictHexStruct),
+  blockHash: StrictHexStruct,
+  blockNumber: StrictHexStruct,
+  gasUsed: StrictHexStruct,
+  transactionHash: StrictHexStruct,
+});
+
+export type GetCallsStatusParams = Infer<typeof GetCallsStatusStruct>;
+export type GetCallsStatusReceipt = Infer<typeof GetCallsStatusReceiptStruct>;
+
+export type GetCallsStatusResult = {
+  status: 'PENDING' | 'CONFIRMED';
+  receipts?: GetCallsStatusReceipt[];
+};
+
+export type GetTransactionReceiptsByBatchIdHook = (
+  batchId: string,
+  req: JsonRpcRequest,
```

Comment:
just curious why the whole req object is needed in the params since it isn't obvious based on the hook name

Reply:
This is standard in the other hooks, as a way to provide any properties set by other middleware such as `networkClientId`.